packages/@aws-cdk/aws-rds/lib/engine.ts

```diff
@@ -0,0 +1,229 @@
+import * as secretsmanager from '@aws-cdk/aws-secretsmanager';
+import { compare } from './private/version';
+import { ParameterGroupFamily } from './props';
+
+abstract class BaseEngine {
+  /**
+   * The engine.
+   */
+  public readonly name: string;
+
+  /**
+   * The single user secret rotation application.
+   */
+  public readonly singleUserRotationApplication: secretsmanager.SecretRotationApplication;
+
+  /**
+   * The multi user secret rotation application.
+   */
+  public readonly multiUserRotationApplication: secretsmanager.SecretRotationApplication;
+
+  protected readonly parameterGroupFamilies?: ParameterGroupFamily[];
```

Comment:
This, and the 4th parameter on the constructor, are only needed on the `DatabaseClusterEngine` class.

Reply:
Yes, but they will be used in later PRs for engines, so I'd rather not remove them.

Reply:
Isn't this a case where we're leaving the state of the codebase in a weird half baked state, when the other PR takes a different turn or plans change?

Would prefer that this PR do the correct thing now and adjust it later as part of the future PR.

Reply:
Isn't this a case where we're leaving the state of the module in a half baked state, if/when the other PR takes a different turn or plans change?

Would prefer that this PR do the correct thing now and adjust it later as part of the future PR.

Reply:
> Isn't this a case where we're leaving the state of the module in a half baked state, if/when the other PR takes a different turn or plans change?

No. This field is not public in `DatabaseInstanceEngine`, so it doesn't leave the module in any different shape than it is now.